static.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const assert = require('assert')
+const path = require('path')
+const fs = require('fs')
+const yaml = require('js-yaml')
+
+module.exports = function (fastify, opts, next) {
+  assert(opts.specification, 'specification is missing in the module options')
+  assert(opts.specification.type, 'specification.type is missing')
+  assert(
+    typeof opts.specification.type === 'string',
+    'specification.type is not a string'
+  )
+  opts.specification.type = opts.specification.type.toLowerCase()
+  assert(
+    ['file'].indexOf(opts.specification.type) !== -1,
+    "specification.type is incorrect, should be one from ['file']"
+  )
+
+  let swaggerObject = {}
+
+  if (opts.specification.type === 'file') {
+    assert(
+      opts.specification.path,
+      'specification.path is missing, should be path to the file'
+    )
+    assert(
+      typeof opts.specification.path === 'string',
+      'specification.path is not a string'
+    )
+    assert(
+      fs.existsSync(path.resolve(opts.specification.path)),
+      `${opts.specification.path} does not exist`
+    )
+    const extName = path.extname(opts.specification.path).toLowerCase()
+    assert(
+      ['.yaml', '.json'].indexOf(extName) !== -1,
+      "specification.path extension name is not supported, should be one from ['.yaml', '.json']"
+    )
```

Comment:
Because of the asynchronous loading of fastify, all those asserts should call `next(err)`.

Reply:
This is a good catch, do you see (or know) if it is possible to achieve this with `assert`? (without me diving into the fastify core code)
Because in the other module, where I actually test the validation, I was forced to refactor it to:
```
  if (!opts.secret) { return next(new Error('fastify-keycloak: secret should be provided')); }
```

if I use `assert(false, next('error'))`, what happens with the exception which is thrown by `assert`?

![image](https://user-images.githubusercontent.com/839290/37281065-07fcef98-25f0-11e8-9fcf-bced037d5fa8.png)

Reply:
You can wrap your asserts with a try/catch block.

Reply:
yes, I've also tried this, but after thought that `if` and `next(new Error())` is better.

ok, thanks for your comments and feedback, I'll try to fix them later this day.

Reply:
refactored to `if` + `return next(new Error())`